python/ray/serve/batching.py

```diff
@@ -137,7 +137,15 @@ async def wait_for_batch(self) -> List[Any]:
             # Add all new arrivals to the batch.
             while len(batch) < max_batch_size and not self.queue.empty():
                 batch.append(self.queue.get_nowait())
-            self.queue_put_event.clear()
+            
+            # Only clear the put event if the queue is empty. If it's not empty
+            # we can start construcing a new batch immediately in the next loop.
+            # The code that puts items into the queue runs on the same event loop
+            # as this code, so there's no race condition between the time we
+            # get objects in the queue (and clear the event) and when objects
+            # get added to the queue.
```

Comment:
This comment is kind of unnecessary because this is standard asyncio concurrency model but better to over communicate than under I suppose :)